tests/unit/test_req_file.py

```diff
@@ -301,6 +301,30 @@ def test_extras_for_editable_url_requirement(self):
         assert req.extras[0] == 'ex1'
         assert req.extras[1] == 'ex2'
 
+    def test_options_and_markers(self):
+        line = 'SomeProject ; python_version=="2.6" --install-option "yo"'
```

Comment:
I went this way, because I think the marker has to be "scoped" to the specifier.   it has to be that way in the shell, so to be consistent, it should be the same in a requirements line.

point taken about the changelog warning.